osquery/tests/integration/tables/programs.cpp

```diff
@@ -16,31 +15,21 @@
 
 namespace osquery {
 
-class programs : public IntegrationTableTest {};
+class ProgramsTest : public IntegrationTableTest {};
 
-TEST_F(programs, test_sanity) {
-  // 1. Query data
-  // QueryData data = execute_query("select * from programs");
-  // 2. Check size before validation
-  // ASSERT_GE(data.size(), 0ul);
-  // ASSERT_EQ(data.size(), 1ul);
-  // ASSERT_EQ(data.size(), 0ul);
-  // 3. Build validation map
-  // See IntegrationTableTest.cpp for avaialbe flags
-  // Or use custom DataCheck object
-  // ValidatatioMap row_map = {
-  //      {"name", NormalType}
-  //      {"version", NormalType}
-  //      {"install_location", NormalType}
-  //      {"install_source", NormalType}
-  //      {"language", NormalType}
-  //      {"publisher", NormalType}
-  //      {"uninstall_string", NormalType}
-  //      {"install_date", NormalType}
-  //      {"identifying_number", NormalType}
-  //}
-  // 4. Perform validation
-  // validate_rows(data, row_map);
+TEST_F(ProgramsTest, test_sanity) {
+  QueryData data = execute_query("select * from programs");
+  ASSERT_GT(data.size(), 0ul);
+  ValidatatioMap row_map = {{"name", NonEmptyString},
+                            {"version", NormalType},
+                            {"install_location", NormalType},
+                            {"install_source", NormalType},
+                            {"language", NormalType},
+                            {"publisher", NormalType},
+                            {"uninstall_string", NormalType},
+                            {"install_date", NormalType},
```

Comment:
Is it legal for the `install_date` to be empty string?

Reply:
it's only windows table and parses registry. It can definitely be removed not sure if it can be not added. In any way, it's not the table problem if value is missing.

Reply:
I consider, even "name", NonEmptyString as flaky but at least thats the most logical one not to be missing.